lib/tasks/performance.rake

```diff
@@ -0,0 +1,229 @@
+
+# frozen_string_literal: true
+
+namespace :perf do
+  desc 'Trim database down to a specific number of patients'
+  task trim: :environment do
+    unless Rails.env == 'development' || ENV['DISABLE_DATABASE_ENVIRONMENT_CHECK']
+      puts 'bundle exec rails perf:populate'
+      raise 'This task is only for use in a development environment' 
+    end
+
+    if ENV['COUNT'].nil? || ENV['COUNT'].to_i <= 0
+      puts "\nMust provide COUNT environment variable to indicate how many patients should be in the DB"
+      puts 'export COUNT=500000'
+      exit 1
+    end
+
+    desired_patient_count = ENV['COUNT'].to_i
+    current_patient_count = Patient.count
+
+    puts "Currently at #{current_patient_count} patients"
+    puts "Script will delete patients until it reaches #{desired_patient_count}\n"
+
+    Patient.uncached do
+      patients = Patient.where('patients.responder_id = patients.id')
+          .includes(
+            :histories,
+            :transfers,
+            :laboratories,
+            :vaccines,
+            :close_contacts,
+            :contact_attempts,
+            assessments: { reported_condition: :symptoms },
+            dependents: [
+              :histories,
+              :transfers,
+              :laboratories,
+              :vaccines,
+              :close_contacts,
+              :contact_attempts,
+              { assessments: { reported_condition: :symptoms } }
+            ]
+          )
+
+      while current_patient_count > desired_patient_count
+        patients.find_in_batches(batch_size: 100) do |group|
+          break unless current_patient_count > desired_patient_count
+          group.each do |patient|
+            break unless current_patient_count > desired_patient_count
+            current_patient_count -= remove_patient(patient)
+            print "\rPatient Count: #{current_patient_count}                   "
+          end
+        end
+      end
+    end
+    puts 'Done!'
+  end
+
+  def remove_patient(patient)
+    removed_patients = 1
+    ActiveRecord::Base.transaction do
```

Comment:
would we be able to use `destroy_all` to automatically remove all the associated records?

https://stackoverflow.com/questions/6698207/delete-all-vs-destroy-all

Reply:
I chose to use `delete_all` because it avoids more SQL queries compared to `destroy_all`

## destroy_all
<img width="1092" alt="Screen Shot 2021-06-30 at 10 29 24 AM" src="https://user-images.githubusercontent.com/24628687/123997835-0b1d8700-d98e-11eb-9761-a692ad3f8743.png">

## delete_all
<img width="676" alt="Screen Shot 2021-06-30 at 10 29 48 AM" src="https://user-images.githubusercontent.com/24628687/123997880-196ba300-d98e-11eb-8d64-4c6830e2df0d.png">

Reply:
good call, thanks for the explanation!